.github/workflows/ci.yml

```diff
@@ -0,0 +1,36 @@
+name: Codecov
+on:
+  push:
+    branches: [ git_workflow ]
+jobs:
+  codecov:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 14.x
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14.x
+      - name: Install dependencies
+        run: npm install
+      - name: Run tests
+        run: npm test 
+      - name: Run coverage
+        run: npm run coverage 
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+           token: ${{ secrets.CODECOV_TOKEN }}
+
+  publish-npm:
+    needs: codecov
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 16
+          registry-url: https://registry.npmjs.org/
+      - run: npm ci
+      - run: npm config set //registry.npmjs.org/:_authToken ${{secrets.NPM_TOKEN}}
+      - run: npm publish
```

Comment:
**question**: won't we need to bump version before publish?

Reply:
while publishing it will detect from package.json